### PR TITLE
Customer service role

### PR DESCRIPTION
**Pull request by Dantemss**
Includes https://github.com/openstax/tutor-server/pull/672

- Added customer service role model and migration
- Admin can add customer service role to users
- Customer service role can access show/index actions in the admin console
- Added model spec
- FE now gets is_customer_service in the user json
- Copied and modified relevant admin controllers and views
- Copied/modified controller/view specs and added some more

### CODE REVIEW

**Comment by Dantemss** [comment]
Do we want to reuse the admin controllers (like in the PR right now), or make separate ones for customer service?

**Comment by jpslav** [comment]
Can you rebase now that 672 is merged?  

I haven't looked at the admin views as a customer service user, but are the views getting complicated with a lot of "if customer_service_user do blah" or "do blah if admin" then we may want to consider dedicated screens for each role.

**Comment by Dantemss** [comment]
I think the only thing that would need a lot of ifs would be the console itself. And the courses `show` page is already completely new (because usually all this course info is only displayed on the `edit` page).

Maybe a separate console, but reuse the other pages then?

**Comment by jpslav** [comment]
Seems reasonable. That way we can also put specific messaging and options all in one place for customer service folks

**Comment by Dantemss** [comment]
OK I decided to just make them separate... Customer support views were simple enough that it wasn't too bad.

**Comment by Dantemss** [comment]
All that FE needs to provide is a Customer Support button or redirect. You can test the control panel at `/customer_support`.

File: app/views/customer_service/targeted_contracts/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<% @page_header = "Targeted Contracts" %>
```
Comment (jpslav) [inline app/views/customer_service/targeted_contracts/index.html.erb]: Don't want customer service folks doing anything with targeted contracts -- you can take these out.

File: config/routes.rb [hunk 2]
```diff
@@ -149,6 +149,27 @@
...
 
+  namespace 'customer_service' do
+    root to: 'console#index'
+
+    resources :courses, only: [:index, :show] do
+      resources :periods, only: [:index], shallow: true
+      resources :students, only: [:index], shallow: true
+    end
+
+    resources :licenses, only: :index
```
Comment (jpslav) [inline config/routes.rb]: I think licenses is something copied from OX Exercises -- can you remove?

Comment (jpslav) [inline config/routes.rb]: here and admin routes

**Comment by jpslav** [comment]
Ok, Dante, don't kill me for this... I think there's too much duplicated code here.  

Can we go the other way so there is only one set of "management" or "console" views that both admins and customer service folks can access, but with admins being the only ones who can see edit capabilities and use those edit capabilities?

The routes namespace can be "console" or "manage" or "management" and the controllers / views can use the same word so we don't have customer service people operating under something named "admin".

While the current approach has some benefits, I fear we're going to get stuck evolving two different consoles and always needing to fix bugs in two places, etc, and it will become an increasing pain in the rear.

Thoughts?

File: config/routes.rb (hunk 2 above)
Comment (Dantemss) [inline config/routes.rb]: ok

File: app/views/customer_service/targeted_contracts/index.html.erb (hunk 1 above)
Comment (Dantemss) [inline app/views/customer_service/targeted_contracts/index.html.erb]: ok

**Comment by Dantemss** [comment]
@jpslav ok I split the shared partials out.

File: app/assets/javascripts/customer_service.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  var tab = window.location.hash;
+  if (tab) {
+    $('a[href="' + tab + '"]').click();
+  }
+});
+
+// === Customer Service/Jobs === //
+$(function(){
+  $(document).on('click', '.filter_job_status', function(e) {
+    var desiredStatus = this.href.replace(/^\S+#/, ''),
```
Comment (jpslav) [inline app/assets/javascripts/customer_service.js]: Can this JS be in a partial?

File: app/assets/stylesheets/customer_service.scss [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  color: red;
+}
+
+#jobs {
+  .completed {
+    display: none;
+  }
+}
+
+#search_by_id {
```
Comment (jpslav) [inline app/assets/stylesheets/customer_service.scss]: Ditto for this jobs-specific CSS (make it shared too)

File: app/controllers/admin/jobs_controller.rb [hunk 5]
```diff
@@ -14,6 +15,7 @@ def show
       end
 
       @page_header = "#{@job.status.titleize} job : #{@job.id}"
```
Comment (jpslav) [inline app/controllers/admin/jobs_controller.rb]: Can you move the @page_headers into the partials?

File: app/controllers/customer_service/courses_controller.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+  private
+
+  def get_schools
+    @schools = SchoolDistrict::ListSchools[]
+  end
+
+  def get_course_details
+    entity_course = Entity::Course.find(params[:id])
+    @course = GetCourseProfile[course: entity_course]
```
Comment (jpslav) [inline app/controllers/customer_service/courses_controller.rb]: I still don't like that these 15 lines are copied.

File: app/assets/javascripts/customer_service.js (hunk 3 above)
Comment (Dantemss) [inline app/assets/javascripts/customer_service.js]: Probably...

File: app/controllers/admin/jobs_controller.rb [hunk 7]
```diff
@@ -14,6 +15,7 @@ def show
       end
```
Comment (jpslav) [inline app/controllers/admin/jobs_controller.rb]: What about moving these other lines into the views too?

File: app/controllers/admin/jobs_controller.rb (hunk 5 above)
Comment (Dantemss) [inline app/controllers/admin/jobs_controller.rb]: I don't think this belongs in partials? It's a page-wide thing to be set by the base view...

File: app/controllers/customer_service/courses_controller.rb (hunk 6 above)
Comment (Dantemss) [inline app/controllers/customer_service/courses_controller.rb]: I can include a module or something

File: app/controllers/admin/jobs_controller.rb (hunk 7 above)
Comment (Dantemss) [inline app/controllers/admin/jobs_controller.rb]: Too much business logic for the views... I'll make another module.

**Comment by Dantemss** [comment]
@jpslav how about now?

**Comment by jpslav** [comment]
Beautiful - thanks Dante

**Pull request merged**